Prepare 1.26.0 (#5424)
open-telemetry/opentelemetry-java: CHANGELOG.md

## Unreleased

This release represents the release candidate ("RC") release for the Logs Bridge API / SDK. In the

> following release, `opentelemetry-api-logs` will be merged into `opentelemetry-api`,

Reviewer: (I initially read "following release" as the release that is described in the following text)

```suggestion
next release, `opentelemetry-api-logs` will be merged into `opentelemetry-api`,
```

`opentelemetry-sdk-logs` will be marked as stable, `opentelemetry-exporter-otlp-logs` will be
merged into `opentelemetry-exporter-otlp`, `opentelemetry-sdk-logs-testing` will be merged
into `opentelemetry-sdk-testing`, `opentelemetry-sdk-extension-autoconfigure` will enable `otlp`

> exporter by default (i.e. `otel.logs.exporter=otlp`). For more details, see tracking

Reviewer: ```suggestion
log exporter by default (i.e. `otel.logs.exporter=otlp`). For more details, see tracking
```

issue [#5340](https://github.com/open-telemetry/opentelemetry-java/issues/5340). NOTE: reminder that
the [Logs Bridge API](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/bridge-api.md)
is _not_ meant for end users. Log appenders use the API to bridge logs from existing log
frameworks (e.g. JUL, Log4j, SLf4J, Logback) into OpenTelemetry. Users configure the Log SDK to
dictate how logs are processed and exported.

`opentelemetry-opentracing-shim` is now stable!

### SDK

> * Create OtelVersion class at build time. Use to resolve artifact version in `Resrouce`.

Reviewer 2: ```suggestion
* Create OtelVersion class at build time. Use to resolve artifact version in `Resource`.
```

> * Create OtelVersion class at build time. Use to resolve artifact version in `Resrouce`.

Reviewer: ```suggestion
* Create OtelVersion class at build time which is used to resolve artifact version in `Resource`.
```

  ([#5365](https://github.com/open-telemetry/opentelemetry-java/pull/5365))

#### Metrics

* Add prototype histogram advice API (i.e. Hints).
  ([#5217](https://github.com/open-telemetry/opentelemetry-java/pull/5217))

#### Logs

* Add LogRecord observed timestamp field.
  ([#5370](https://github.com/open-telemetry/opentelemetry-java/pull/5370))
* Remove log record timestamp default.
  ([#5374](https://github.com/open-telemetry/opentelemetry-java/pull/5374))
* Align BatchLogRecordProcessor defaults with specification.
  ([#5364](https://github.com/open-telemetry/opentelemetry-java/pull/5364))
* Rename setEpoch to setTimestamp.
  ([#5368](https://github.com/open-telemetry/opentelemetry-java/pull/5368))

### Exporters

> * Add skaffolding for low level exporter TLS API.

Reviewer 2: Typo?
```suggestion
* Add scaffolding for low level exporter TLS API.
```
